packages/manager/apps/pci-gateway/src/api/hooks/useGateway.ts

```diff
@@ -41,3 +43,39 @@ export const useAggregatedGateway = (
     };
   }, [isLoading, error, gateways, pagination, filters]);
 };
+
+type RemoveGatewayProps = {
+  projectId: string;
+  gatewayId: string;
+  region: string;
+  onError: (cause: Error) => void;
+  onSuccess: () => void;
+};
+
+export const useDeleteGateway = ({
+  projectId,
+  gatewayId,
+  region,
+  onError,
+  onSuccess,
+}: RemoveGatewayProps) => {
+  const mutation = useMutation({
+    mutationFn: async () => {
+      return deleteGateway(projectId, region, gatewayId);
+    },
+    onError,
+    onSuccess: async () => {
+      await queryClient.invalidateQueries({
```

Comment:
here since you delete an item, no need to invalidate the query containing the list of gateways, that will trigger another call, may be it's more convenient to update its content removing the gateway you just removed, that means one less http call, unless you really need to do it